### ISSUE TITLE
How to increase performance for VirtualTable component?

### ISSUE THREAD

**Issue by drilus** [DevExpress/devextreme-reactive#2131]
- [x] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [x] React Grid
- [ ] React Chart
- [ ] React Scheduler

### Description
I'm new to React and picked a project that would be practical. I really like the design and customization options on the dev-extreme components. However, I'm having performance issues with the VirtualTable component. Scrolling horizontally and vertically are both laggy. I'm using both column and row virtualization. I'm loading a 9MB csv file that has 32 columns and 18837 rows.

If anyone is willing to help me out fix the performance issues I'd appreciate it. I thought I've followed the examples and documentation but I'm obviously missing something.

I have created a codesandbox if anyone wants to take a look:
https://codesandbox.io/s/github/drilus/parse_csv

### Environment
```javascript
  "dependencies": {
    "@devexpress/dx-react-core": "^1.11.1",
    "@devexpress/dx-react-grid": "^1.11.1",
    "@devexpress/dx-react-grid-material-ui": "^1.11.1",
    "@fortawesome/fontawesome-svg-core": "^1.2.19",
    "@fortawesome/free-solid-svg-icons": "^5.9.0",
    "@fortawesome/react-fontawesome": "^0.1.4",
    "@material-ui/core": "^4.1.3",
    "@material-ui/icons": "^4.2.1",
    "console-feed": "^2.8.8",
    "papaparse": "^5.0.0",
    "react": "^16.8.6",
    "react-bootstrap": "^1.0.0-beta.9",
    "react-dock": "^0.2.4",
    "react-dom": "^16.8.6",
    "react-scripts": "3.0.1",
    "react-virtualized": "^9.21.1"
  }
```

**Comment by ushkal** [comment]
Hi @drilus ,
 
We've researched your project but haven't found any performance bottlenecks in the code. However we found an issue when a grid header was unnecessarily updated. We'll fix it soon. 
Here are common recommendations regarding performance:
- ensure you are deploying a production build of an application
- in a virtual table, rows are added and removed on scrolling. For this reason a stripped table doesn't work as intended. It doesn't influence performance but affects user experience. Refer to[ this thread](https://github.com/DevExpress/devextreme-reactive/issues/1407) to find a solution.
